src/partition/repart.c

```diff
@@ -407,6 +462,14 @@ static bool context_drop_one_priority(Context *context) {
 
                 p->dropped = true;
                 log_info("Can't fit partition %s of priority %" PRIi32 ", dropping.", p->definition_path, p->priority);
+
+                if (p->verity != VERITY_OFF) {
+                        Partition *dp;
+
+                        assert_se(dp = find_verity_sibling(context->partitions, p));
+                        dp->dropped = true;
+                        log_info("Also dropping sibling verity partition %s", dp->definition_path);
+                }
```

Comment:
hmm, this should take the priority into consideration.

or maybe better: refuse a configuration where the verity data partition has a different priority than the hash partition already when parsing.

And then add comment here saying that it's safe to drop the sibling regardless of priority since we checked priority matching already before.